connectors/database/postgres.go

```diff
@@ -0,0 +1,39 @@
+package sql
+
+import (
+	"fmt"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+func InitPostgresClient(dbConfig map[string]string) *gorm.DB {
+	host := dbConfig["host-v3"]
+	username := dbConfig["username-v3"]
+	password := dbConfig["password-v3"]
+	port := dbConfig["port-v3"]
+	dbname := dbConfig["database-v3"]
+
+	if host == "" {
+		return nil
+	}
+
+	dsn := fmt.Sprintf("host=%s user=%s password=%s dbname=%s port=%s", host, username, password, dbname, port)
```

Comment:
Add validation for all required database configuration parameters (e.g., `username-v3`, `password-v3`, `port-v3`, `database-v3`) before constructing the DSN. This ensures that the function fails early with a clear error message if any required parameter is missing. [important]